biz.aQute.repository/src/aQute/bnd/repository/maven/provider/IndexFile.java

```diff
@@ -194,23 +194,25 @@ boolean refresh() throws Exception {
 			last = System.currentTimeMillis();
 			return true;
 		}
-		return descriptors.values()
-				.stream()
-				.filter(descriptor -> {
-					try {
-						if ((descriptor.promise != null) && descriptor.promise.isDone() && (descriptor.promise.getFailure() == null)) {
-							File f = descriptor.promise.getValue();
-							if ((f != null) && f.isFile() && (f.lastModified() != descriptor.lastModified)) {
-								updateDescriptor(descriptor, f);
-								return true;
-							}
-						}
-					} catch (InvocationTargetException | InterruptedException e) {
-						throw Exceptions.duck(e);
+
+		boolean updated = false;
```

Comment:
This is a pointless change and adds no value. It was a for loop before I changed it to streams like the rest of the descriptors operations in this source file.

Using a for loop imposes an iteration order that is not required. Using streams would allow the streams implementation to potentially optimize the iteration order in the future.

Reply:
> This is a pointless change and adds no value.

I disagree.

> It was a for loop before I changed it to streams like the rest of the descriptors operations in this source file.

So? Every other use of streams in the file is concise and obvious and has no bearing on this usage.

> Using a for loop imposes an iteration order that is not required. Using streams would allow the streams implementation to potentially optimize the iteration order in the future.

I would err on the side of code clarity over _potential_ optimization. If the performance of this particular method is that much of a consideration then we should explicitly parallelize it and not just cross our fingers that the JVM will do it for us at the expense of readable code.

Reply:
I guess we will have to disagree here. I don't disagree with the other commits but this one I disagree with.

I am not sure how a for loop (external iteration) has more clarity than stream processing (internal iteration).

Reply:
> I am not sure how a for loop (external iteration) has more clarity than stream processing (internal iteration).

I think were it using `forEach` (with an external flag) instead of `filter`/`count` it wouldn't bother me as much. As it is, the specific count of items that satisfy the filter predicate is not relevant beyond "is it 0 or is it greater than 0" but we can't use a short circuiting terminal operation because we need to ensure that we process the filter predicate for all items. Simple!

On the other hand, what the for loop is doing is obvious.

Reply:
Also, I've rebased that commit away.